src/SessionSetup/RetrieveInstancesWidgetTest.cpp

```diff
@@ -51,17 +119,85 @@ class RetrieveInstancesWidgetTest : public testing::Test {
 
  protected:
   void SetUp() override {
+    QCoreApplication::setOrganizationName(kOrganizationName);
+    QCoreApplication::setApplicationName(kApplicationName);
+    QSettings settings;
+    settings.clear();
+
     ASSERT_NE(filter_line_edit_, nullptr);
     ASSERT_NE(all_check_box_, nullptr);
     ASSERT_NE(project_combo_box_, nullptr);
     ASSERT_NE(reload_button_, nullptr);
   }
-  MockGgpClient mock_ggp_;
+
+  void VerifyAllElementsAreEnabled() {
+    EXPECT_TRUE(filter_line_edit_->isEnabled());
+    EXPECT_TRUE(all_check_box_->isEnabled());
+    EXPECT_TRUE(project_combo_box_->isEnabled());
+    EXPECT_TRUE(reload_button_->isEnabled());
+  }
+
+  void VerifyOnlyReloadIsEnabled() {
+    EXPECT_FALSE(filter_line_edit_->isEnabled());
+    EXPECT_FALSE(all_check_box_->isEnabled());
+    EXPECT_FALSE(project_combo_box_->isEnabled());
+    EXPECT_TRUE(reload_button_->isEnabled());
+  }
+
+  void VerifyLastLoadingReturnedInstanceList(const QVector<Instance>& instances) {
+    QList<QVariant> arguments = loading_successful_spy_.last();
+    ASSERT_EQ(arguments.count(), 1);
+    ASSERT_TRUE(arguments.at(0).canConvert<QVector<Instance>>());
+
+    EXPECT_EQ(arguments.at(0).value<QVector<Instance>>(), instances);
+  }
+
+  void VerifyProjectComboBoxHoldsData(const QVector<Project>& projects,
```

Comment:
Nit: This tests a lot more than just "does the project combo hold data" as the name suggests. Maybe `VerifyProjectComboBoxData`?

Reply:
done